### PR TITLE
Implement chat bot backend & frontend

### PR DESCRIPTION
**Pull request by lemilonkh**
- Adds support for HuggingFace (OpenAssistant model) and OpenAI (not yet fully tested - need a valid API key)
- Implements UI design
- Adds it to dashboard

### CODE REVIEW

File: app/src/app/api/v0/chat/route.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  return new StreamingTextResponse(stream);
+}
+
+export const POST = apiHandler(async (req: Request) => {
+  const { messages } = chatRequest.parse(await req.json());
+  if (!messages[0].content.startsWith("You are a")) {
+    messages.unshift({
+      role: process.env.CHAT_PROVIDER == "huggingface" ? "user" : "system",
+      content:
+        'You are a climate assistant for creating GPC climate inventories using CityCatalyst, an open source tool for creating climate inventories by Open Earth Foundation. You try to be as helpful as possible when answering the user\'s questions about their inventory or any climate science or data science related questions. Try to be as scientific as possible.',
```
[the remark this answers is not in the document]
Comment (cephaschapa) [reply app/src/app/api/v0/chat/route.ts:89]: Also there cases when chatgpt or huggingface has an outage, how do we approach that?

File: app/src/components/ChatBot/chat-bot.tsx [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    ],
+  });
+  const { copyToClipboard, isCopied } = useCopyToClipboard({});
+  const { formRef, onKeyDown } = useEnterSubmit();
+  const messagesWrapperRef = useRef<HTMLDivElement>(null);
+
+  const userStyles = "rounded-br-none";
+  const botStyles = "rounded-bl-none";
+  const suggestions = [
+    {
```
Comment (cephaschapa) [inline app/src/components/ChatBot/chat-bot.tsx:80]: Would be awesome to have these translated in future if possible, curious to see how that works 😏

**Review by cephaschapa — APPROVED** [review]

File: app/src/app/api/v0/chat/route.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lemilonkh) [reply app/src/app/api/v0/chat/route.ts:89]: Multi-language could be eventually possible, this means we likely need to use different LLM models for those specific languages.
With outages we can show user feedback for now, or try to switch to a different model if one is unavailable if necessary. Would be handled in a future PR though imo.

File: app/src/components/ChatBot/chat-bot.tsx (hunk 2 above)
Comment (lemilonkh) [reply app/src/components/ChatBot/chat-bot.tsx:80]: These will be generated by the LLM in the future as well based on the MVP prompts for this.
So once we support different languages in the LLM we can also have the prompt suggestions in those languages :+1:

**Pull request merged**